src/averageAges.js

```diff
@@ -55,7 +75,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let amount = 0;
+  const result = people.reduce((summ, person) => {
+    const temp = people.find(el => el.name === person.mother);
+
+    if (person.mother
+    && ((onlyWithSon && person.sex === 'm')
+    || (!onlyWithSon)) && temp) {
+      amount++;
```

Comment:
make it simple, it's hard to understand what is going on here
use something like that 
```
const children = people.filter(person =>
    onlyWithSon
      ? 
      : 
 
const ageDiff = children.map(child =>
    child.born - mother.born);
    
 return (ageDiffSum / ageDiff.length)
```